### ISSUE TITLE
test failure on system-3.8 test

### ISSUE THREAD

**Issue by dandhlee** [googleapis/google-resumable-media-python#214]
Hey folks, I'm noticing a test failure that's preventing PRs to be submitted. Here's the relevant stack trace (I think)

```
tests_async/system/requests/test_upload.py ................E

==================================== ERRORS ====================================
__ ERROR at teardown of TestResumableUploadUnknownSize.test_interleave_writes __
Traceback (most recent call last):
  File "/tmpfs/src/github/google-resumable-media-python/.nox/system-3-8/lib/python3.8/site-packages/pytest_asyncio/plugin.py", line 139, in finalizer
    loop.run_until_complete(async_finalizer())
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 591, in run_until_complete
    self._check_closed()
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 508, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
=============================== warnings summary ===============================
.nox/system-3-8/lib/python3.8/site-packages/google/auth/transport/_aiohttp_requests.py:196
  /tmpfs/src/github/google-resumable-media-python/.nox/system-3-8/lib/python3.8/site-packages/google/auth/transport/_aiohttp_requests.py:196: DeprecationWarning: Inheritance class AuthorizedSession from ClientSession is discouraged
    class AuthorizedSession(aiohttp.ClientSession):

tests_async/system/requests/test_download.py:56
  /tmpfs/src/github/google-resumable-media-python/tests_async/system/requests/test_download.py:56: DeprecationWarning: Inheritance class CorruptingAuthorizedSession from ClientSession is discouraged
    class CorruptingAuthorizedSession(tr_requests.AuthorizedSession):

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
ERROR tests_async/system/requests/test_upload.py::TestResumableUploadUnknownSize::test_interleave_writes
=================== 39 passed, 2 warnings, 1 error in 29.99s ===================
Fatal error on SSL transport
protocol: <asyncio.sslproto.SSLProtocol object at 0x7fde26e95a90>
transport: <_SelectorSocketTransport closing fd=11>
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/asyncio/selector_events.py", line 910, in write
    n = self._sock.send(data)
OSError: [Errno 9] Bad file descriptor

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/asyncio/sslproto.py", line 685, in _process_write_backlog
    self._transport.write(chunk)
  File "/usr/local/lib/python3.8/asyncio/selector_events.py", line 916, in write
    self._fatal_error(exc, 'Fatal write error on socket transport')
  File "/usr/local/lib/python3.8/asyncio/selector_events.py", line 711, in _fatal_error
    self._force_close(exc)
  File "/usr/local/lib/python3.8/asyncio/selector_events.py", line 723, in _force_close
    self._loop.call_soon(self._call_connection_lost, exc)
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 719, in call_soon
    self._check_closed()
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 508, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
Unclosed client session
client_session: <google.auth.transport._aiohttp_requests.AuthorizedSession object at 0x7fde27a1b3d0>
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x7fde26f47ca0>, 827.26540656)]']
connector: <aiohttp.connector.TCPConnector object at 0x7fde27a1beb0>
sys:1: RuntimeWarning: coroutine 'pytest_fixture_setup.<locals>.wrapper.<locals>.finalizer.<locals>.async_finalizer' was never awaited
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
Unclosed client session
client_session: <google.auth.transport._aiohttp_requests.AuthorizedSession object at 0x7fde26e14d30>
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x7fde26ec4fa0>, 856.091262762)]']
connector: <aiohttp.connector.TCPConnector object at 0x7fde26fc77c0>
Fatal error on SSL transport
protocol: <asyncio.sslproto.SSLProtocol object at 0x7fde26e1bd90>
transport: <_SelectorSocketTransport closing fd=6>
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/asyncio/selector_events.py", line 910, in write
    n = self._sock.send(data)
OSError: [Errno 9] Bad file descriptor

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/asyncio/sslproto.py", line 685, in _process_write_backlog
    self._transport.write(chunk)
  File "/usr/local/lib/python3.8/asyncio/selector_events.py", line 916, in write
    self._fatal_error(exc, 'Fatal write error on socket transport')
  File "/usr/local/lib/python3.8/asyncio/selector_events.py", line 711, in _fatal_error
    self._force_close(exc)
  File "/usr/local/lib/python3.8/asyncio/selector_events.py", line 723, in _force_close
    self._loop.call_soon(self._call_connection_lost, exc)
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 719, in call_soon
    self._check_closed()
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 508, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
```

Please help! Thank you 😄

**Comment by cojenco** [comment]
Hi @dandhlee! Thanks for filing this. We're looking into it.

**Comment by cojenco** [comment]
This is due to an [issue in the pytest-asyncio package](https://github.com/pytest-dev/pytest-asyncio/issues/209). Thanks to @crwilcox for pair debugging. Submitting a fix to unblock PR submissions. Opening another issue to track the pytest-asyncio issue.
